flink-staging/flink-gelly/src/main/java/org/apache/flink/graph/Graph.java

```diff
@@ -326,6 +328,34 @@ public ExecutionEnvironment getContext() {
 	}
 
 	/**
+	 * @return a triplet DataSet consisting of (srcVertexId, trgVertexId, srcVertexValue, trgVertexValue, edgeValue)
+	 */
+	public DataSet<Triplet<K, VV, EV>> getTriplets() {
+		return this.getVertices().join(this.getEdges()).where(0).equalTo(0)
+				.with(new FlatJoinFunction<Vertex<K, VV>, Edge<K, EV>, Tuple4<K, K, VV, EV>>() {
+
+					@Override
+					public void join(Vertex<K, VV> vertex, Edge<K, EV> edge, Collector<Tuple4<K, K, VV, EV>> collector)
+							throws Exception {
```

Comment:
We should run some experiments and check whether this creates a deadlock. It's the same join pattern that created problems with #408.